testing/sdk_plan.py

```diff
@@ -201,13 +203,22 @@ def get_plans_on_failure(request):
     """
     yield
     if sdk_utils.is_test_failure(request):
-        for plan_name in list_plans(framework_name):
-            plan = get_plan(framework_name, plan_name)
-            if not plan:
-                log.error('Unable to fetch {} plan for {}'.format(framework_name, plan_name))
-                continue
-            out_path = os.path.join(
-                sdk_utils.get_test_log_directory(request.node),
-                '{}_plan.txt'.format(plan_name))
-            with open(out_path, 'w') as f:
-                f.write(json.dumps(plan, indent=2))
+        try:
+            log.info('Fetching plans from {}...'.format(framework_name))
+            plan_names = list_plans(framework_name, 5)
+            log.info('Plans for {}: {}'.format(framework_name, plan_names))
+            for plan_name in plan_names:
+                log.info('Fetching {} plan: {}'.format(framework_name, plan_name))
+                plan = get_plan(framework_name, plan_name, 5)
+                if not plan:
+                    log.error('Unable to fetch {} plan for {}'.format(framework_name, plan_name))
+                    continue
+                out_path = os.path.join(
+                    sdk_utils.get_test_log_directory(request.node),
+                    '{}_plan.txt'.format(plan_name))
```

Comment:
should the output be `.txt`? Are we not writing a `.json` file? We could also use:
```python
json.dump(plan, f, indent=2)
```
to write directly to the file (unless the contents are important from a logging perspective?

Reply:
It'd be nice to keep the following logging of the number of bytes. Checked if `json.dump()` returned the number of bytes written, but it didn't.